metricbeat/module/elasticsearch/node_stats/data_xpack.go

```diff
@@ -187,18 +187,15 @@ func eventsMappingXPack(r mb.ReporterV2, m *MetricSet, info elasticsearch.Info,
 		return errors.Wrap(err, "failure parsing Elasticsearch Node Stats API response")
 	}
 
-	// Normally the nodeStruct should only contain one node. But if _local is removed
-	// from the path and Metricbeat is not installed on the same machine as the node
-	// it will provid the data for multiple nodes. This will mean the detection of the
-	// master node will not be accurate anymore as often in these cases a proxy is in front
-	// of ES and it's not know if the request will be routed to the same node as before.
+	// TODO: call GET _nodes/_master?filter_path=nodes.*.name to figure out the ID of the master node
```

Comment:
I guess this TODO is done by `GetMasterNodeID`

Reply:
Removed in 34cd849.